app/models/megaphone/episode.rb

```diff
@@ -237,19 +280,27 @@ def handle_response(api_response)
 
     # update delivery status after a create or update
     def update_delivery_status
-      # if there is audio
-      if feeder_episode.complete_media? && !has_media_version?
+      # if there is not audio yet, we're all done
+      if !feeder_episode.complete_media?
+        delivery_status(true).mark_as_delivered!
+      # if the audio doesn't match, either it was just uploaded, or needs it
+      elsif !has_media_version?
         # if there's audio and we just uploaded it successfully, set attr, then check status
         if background_audio_file_url
-          attrs = source_attributes.merge(uploaded: true, delivered: false)
+          attrs = source_attributes.merge(
+            enclosure_url: background_audio_file_url,
+            uploaded: true,
+            delivered: false
+          )
           feeder_episode.update_episode_delivery_status(:megaphone, attrs)
-        # if versions don't match, and we didn't upload, it isn't delivered
+        # if versions don't match, and we didn't upload, it isn't uploaded or delivered
         else
           delivery_status(true).mark_as_not_delivered!
         end
-      # or if there's not audio yet or it didn't change
       else
-        delivery_status(true).mark_as_delivered!
+        # media is complete and has the right version, that's uploaded!
+        # next pass through check_audio! should mark it delivered if mp matches and is complete
+        delivery_status(true).mark_as_uploaded!
```

Comment:
this is a fix - once there is audio, only `check_audio!` should mark the episode as `delivered`, not here in updating the status after a create or update api call.